src/components/radio-button/radio-button-group/radio-button-group.test.tsx

```diff
@@ -0,0 +1,292 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { RadioButton, RadioButtonGroup } from "..";
+import CarbonProvider from "../../carbon-provider";
+import Logger from "../../../__internal__/utils/logger";
+import { mockMatchMedia } from "../../../__spec_helper__/__internal__/test-utils";
+
+test("logs a deprecation warning for uncontrolled behaviour", () => {
+  const loggerSpy = jest
+    .spyOn(Logger, "deprecate")
+    .mockImplementation(() => {});
+  render(
+    <RadioButtonGroup name="group">
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  expect(loggerSpy).toHaveBeenCalledWith(
+    "Uncontrolled behaviour in `Radio Button` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+  expect(loggerSpy).toHaveBeenCalledTimes(1);
+});
```

Comment:
**suggestion**: restoring the logger's original behavior should prevent the console from being accidentally masked in other tests:

```suggestion
    loggerSpy.mockRestore();
});
```